src/common/hooks/use-setup-tx.ts

```diff
@@ -2,54 +2,64 @@ import { useEffect, useState } from 'react';
 import { useRecoilValue } from 'recoil';
 import { usePrevious } from '@stacks/ui';
 
-import { requestTokenStore } from '@store/transaction';
-import { currentAccountStxAddressStore } from '@store/wallet';
+import { requestTokenStore, transactionPayloadStore } from '@store/transaction';
+import { currentAccountStxAddressStore, hasRehydratedVaultStore } from '@store/wallet';
 
 import { useAccountSwitchCallback } from '@common/hooks/callbacks/use-account-switch-callback';
 import { usePostConditionsCallback } from '@common/hooks/callbacks/use-post-conditions-callback';
 import { useNetworkSwitchCallback } from '@common/hooks/callbacks/use-network-switch-callback';
 import { useDecodeRequestCallback } from '@common/hooks/callbacks/use-decode-request-callback';
 
 export const useSetupTx = () => {
+  const hasRehydratedVault = useRecoilValue(hasRehydratedVaultStore);
```

Comment:
Can you remind me what the `vault` is? This is key storage?

Reply:
The vault is the reducer/key storage for the extension, in the background script.